.github/workflows/reuse.yml

```diff
@@ -0,0 +1,254 @@
+    secrets:
+      BALENACLOUD_API_KEY:
+        required: false
+      BALENAMACHINE_API_KEY:
+        required: false
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Get Leviathan checked out
+      uses: actions/checkout@v4
+      with:
+        repository: balena-os/leviathan
+        submodules: true
+        # path: ./leviathan-workspace
+        # The branch, tag or SHA to checkout. When checking out the repository that
+        # triggered a workflow, this defaults to the reference or SHA for that event.
+        # Otherwise, uses the default branch.
+        ref: ${{ inputs.LEVIATHAN_REF }}
+
+    # - name: Set up Docker Buildx
+    #   uses: docker/setup-buildx-action@v3
+    # Don't think, I need this since docker-compose uses only buildkit which we enabled by default in Makefile
+
+    - name: Build it, and they might come
+      shell: bash
+      env:
+        WORKER_TYPE: ${{ inputs.WORKER_TYPE }}
+        # SECUREBOOT: ${{ inputs.SECUREBOOT }}
+      run: |
+        make config
+        make build
+        docker image ls -a
+
+    - name: Save the images
+      shell: bash
+      run: |
+        docker save --output /tmp/corelatest.tar leviathan_core:latest 
+        docker save --output /tmp/clientlatest.tar leviathan_client:latest 
+
+    - name: Upload Core artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: corelatest.tar
+        path: /tmp/corelatest.tar
+        compression-level: 7
+
+
+    - name: Upload Client artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: clientlatest.tar
+        path: /tmp/clientlatest.tar
+        compression-level: 7
+  
+
+  test:
+    runs-on: ubuntu-22.04
+    needs: build
+    steps:
+    - name: Download artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: clientlatest.tar
+        path: /tmp
+
+    - name: Download artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: corelatest.tar
+        path: /tmp
+
+    - name: Load image
+      shell: bash
+      run: |
+        docker image ls -a
+        docker load --input /tmp/corelatest.tar
+        docker load --input /tmp/clientlatest.tar
+        docker image ls -a
+
+    - name: Fetching code to be tested
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Get Leviathan checked out
+      uses: actions/checkout@v4
+      with:
+        repository: balena-os/leviathan
+        submodules: true
+        path: ./leviathan
+        # The branch, tag or SHA to checkout. When checking out the repository that
+        # triggered a workflow, this defaults to the reference or SHA for that event.
+        # Otherwise, uses the default branch.
+        ref: ${{ inputs.LEVIATHAN_REF }}
+
+
+    # - name: Fetching the Private Contracts
+    #   uses: actions/checkout@v4
+    #   with:
+    #     repository: balena-io/private-contracts
+    #     path: './leviathan/core/private-contracts'
```

Comment:
contracts should already be checked out with the device repo

Reply:
Not the private ones i think?

Reply:
Private contracts are checked out separately in the jenkins config - not included in leviathan as a submodule.

Reply:
Yeah... this should be an easy fix. Just need to find out who's SSH key is in Jenkins and then probably give balena-os org access to that secret which can be inherited by this reusable workflow

Reply:
> Private contracts are checked out separately in the jenkins config - not included in leviathan as a submodule.

yes, but only because in Jenkins we don't have a clone of the device repo, and private device types must include private contracts as a submodule

remember that the repo this will be running on is a device repo, so the secrets to get the private contracts must already be present via the Flowzone auth